apis/sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/BuiltInFunctionResource.java

```diff
@@ -112,7 +112,7 @@ public class BuiltInFunctionResource {
         @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
       })
   @POST
-  @Path("{collection:\\w+}/{document-id}/{document-path:.*}/function")
+  @Path("{collection:\\w+}/{document-id: [^/]+}{slash: /?}{document-path: (.+)?}/function")
```

Comment:
OK, but we still have to fix the micrometer template on this line:

https://github.com/stargate/stargate/blob/2384046a394613eb2d9a6d4dcea2e7712eea4f5f/apis/sgv2-docsapi/src/main/resources/application.yaml#L111

Seems like only the right hand side has to be updated to the new path spec from that `@Path`.. The best way to test this is to hit that endpoint without authentication and see what's get counted in the `/metrics`.. And then hit with the token and assure both calls are having same path tag..

Failing to do this we are risking of a metrics explosion, it's very high risk imo.. Please let me know if you have questions!

Reply:
I feel like the left side needs to be changed, too - no? Because `/v2/namespaces/X/collections/Y/documentId/function` doesn't match the left side regex?